freedomofpress/securedrop-sdk
# support seen endpoint

# Description

Fixes https://github.com/freedomofpress/securedrop-sdk/issues/132

Support for using the updated SDK and endpoints has partial support in the client in a working branch. We are going to use this client to test that the SDK changes in this PR are working. Remember that seen/unseen is still not working completely in the client and that there will be things to iron out, so we just want to pay attention in the test plan to the SDK working as far as sending seen data back and forth between client and server as a pass-through.

# Test Plan
## 1. Test updated /replies and /submissions endpoints
Test that the existing endpoints are retrieving `seen` data
1. from `develop`, run dev server: `NUM_SOURCES=0 make dev`
2. in a different tab, checkout `add-seen-tables` branch in the client repo
3. `pip uninstall securedrop-sdk`
4. `pip install git+https://github.com/freedomofpress/securedrop-sdk@support-seen-endpoint#egg=securedrop-sdk` (sdk from this pr branch that you are testing)
5. `rm -rf ~/.securedrop_client`
7. `LOGLEVEL=debug ./run.sh --sdc-home ~/.securedrop_client`
6. log in as `journalist` and don't click anything yet!
7. in a different tab, run: `sqlite3 ~/.securedrop_client/svs.sqlite`
8. run `select * from seen_replies;`
- [ ] confirm that there are no results 
9. run `select * from seen_messages;`
- [ ] confirm that there are no results 
10. log in a new source, add a file and message submission
11. log in as a journalist, add a reply to that source
12. download the file and message (to mark them as read) from the JI
13. wait for the client sync to pull in the new reply, message, and file
14. back to your sqlite3 tab, run `select * from seen_replies;` (sometimes you need to wait a minute for the db to be updated)
- [ ] confirm that the new reply is now marked as seen (since the sender of the reply automatically sees it)
15. run `select * from seen_messages;`
- [ ] confirm that the new message is now marked as seen
16. run `select * from seen_files;`
- [ ] confirm that the new file is now marked as seen
(continue on to test 2 below)

## 2. Test new endpoint /seen
1. as the same source, send another file and message submission
2. in the client, click on the source in the menu to trigger the call to the SDK's seen api
3. in a new tab, run `docker exec -it securedrop-dev-0 bash` to connect to the dev server
4. run: `sqlite3 /var/lib/securedrop/db.sqlite`
5. run `select * from seen_messages;`
- [ ] confirm that you see all the source messages in the results (since you clicked on that source)
6. run `select * from seen_files;`
- [ ] confirm that you see all the source files in the results (since you clicked on that source)
